packages/suite/src/middlewares/wallet/coinjoinMiddleware.ts

```diff
@@ -133,18 +144,13 @@ export const coinjoinMiddleware =
         }
 
         if (action.type === SUITE.TOR_STATUS) {
-            const state = api.getState();
-            const deviceStatus = selectDeviceState(state);
-            const isDeviceConnected = deviceStatus === 'connected';
-            const isAccountOutOfSync = getIsCoinjoinOutOfSync(state.wallet.selectedAccount);
-            const isCoinjoinResumeAllowed = !isAccountOutOfSync || isDeviceConnected;
-
             if (['Disabling', 'Disabled', 'Error'].includes(action.payload)) {
                 api.dispatch(coinjoinAccountActions.pauseInterruptAllCoinjoinSessions());
             }
             // We restore sessions that were interrupted when successfully Enabled if
             // there is not any other condition blocking coinjoin resume.
-            if (action.payload === 'Enabled' && isCoinjoinResumeAllowed) {
+            const state = api.getState();
+            if (action.payload === 'Enabled' && !checkIsCoinjoinResumeBlocked(state)) {
```

Comment:
At this point the actions are propagated to reducers so it is likely that the next state is already applied but it does not have to be always like that, there could be some race conditions. 
So it could happen that when this run in the state Tor is still disable and `isCoinJoinBlockedByTor` will return true.

Reply:
I think, as long as this is run after the `next()` call, there should be no such issue because `updateTorStatus' is just a simple synchronous action. Or am I wrong?